python/pyarrow/tests/test_plasma.py

```diff
@@ -390,6 +390,12 @@ def __init__(self, val):
             object_id, -1, serialization_context=serialization_context)
         assert result.val == val.val
 
+    def test_store_np_matrix(self):
+        m = np.matrix(np.random.randn(10, 4))
+        object_id = self.plasma_client.put(m)
+        buf = self.plasma_client.get(object_id)
+        assert (buf == m).all()
```

Comment:
Will this catch errors if the dtype changes?

Reply:
It wouldn't e.g. for `int` and `float`. I've added a `dtype` comparison.